### ISSUE TITLE
t.web.util.redirectTo should reject unicode URLs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5236]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#5236|
|Type|enhancement|
|Created|2011-08-23 07:21:52Z|

If you have something like this in your `render`:

`return redirectTo(url, request)`

and you pass a unicode object as the `url`, you get a rather mysterious error page from twisted.web (with the redirection headers set!):

` "500 - Request did not return a string"`

`redirectTo` should really just throw `TypeError` if you pass in a `unicode` object.

Attachments:

* [my-twisted-patch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3db/3db96f61d66b872fca548a19b248be5081dca824/d5bf14bfe4ce8fc7b9256a069dbe396aeaef28e5.patch) (1226 bytes) - added by moijes12 on 2011-12-22 05:43:20Z - Check added to function twisted.web.util.redirectTo so that it raises TypeError if Unicode object passed in URL. Unit test file test_util is also updated with unit test.Comments and docstring are added and test has passed.

<details><summary>Searchable metadata</summary>

```
trac-id__5236 5236
type__enhancement enhancement
reporter__ivank ivank
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1314084112000000 1314084112000000
changetime__1324566602000000 1324566602000000
version__None None
owner__exarkun exarkun
cc__jknight cc__moijes12
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  This looks pretty good.  A few simple points:

  1. We generally don't annotate changes with a reference to the ticket they're for.  `svn blame` does an okay job of pointing out why certain code exists, when someone wants to know.
  1. `redirectTo` should have had a docstring already; since it didn't, and it's being changed, it should really be given one now. :)
  1. There should also be a news fragment describing the change.  This seems worthy of a feature fragment, since it is improving the behavior of a public API.

These are pretty minor points, so I'm happy to make the changes myself and then apply the result.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_4"></a>|@moijes12 commented|
|-|-|

Is there anything more that I should do here ?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [33400]) Reject unicode locations early in twisted.web.util.redirectTo

Author: moijes12
Reviewer: exarkun
Fixes: [#5236](https://github.com/twisted/twisted/issues/5236)

Raise `TypeError` from `redirectTo` if the supplied URL is `unicode` instead of
`str`.  This replaces the old behavior of returning a `unicode` string in this
case which would later cause a 500 response when lower-level parts of Twisted Web
rejected the non-`str` response body string.
